platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/UnsignedEvent.java

```diff
@@ -153,7 +153,7 @@ public UnsignedEvent(
                 HapiUtils.asTimestamp(timeCreated),
                 this.allParents.stream()
                         .map(ed -> new com.hedera.hapi.platform.event.EventDescriptor(
-                                ed.getHash().getBytes(), ed.getCreator().id(), ed.getGeneration(), ed.getBirthRound()))
+                                ed.getHash().getBytes(), ed.getCreator().id(), ed.getBirthRound(), ed.getGeneration()))
```

Comment:
As commented on the GitHub issue separately, it would be ideal to avoid using constructors and instead switch to using builders available in every PBJ model, which could help prevent bugs such as this one.

Reply:
Perhaps the reason this was not done was to avoid two memory allocations for each event

Reply:
Sounds reasonable, provided we have some benchmark data to support this.

Reply:
@OlegMazurov shared that even without the builder, since we now are double hashing events (block stream redaction purposes) the allocation rate jumped significantly, but not enough to affect overall performance. That being the case, we want to be mindful of the allocations.

Reply:
@OlegMazurov also noted in a separate PR that memory allocations for short-lived objects are okay for the performance. So I'm not sure if that number alone is sufficient to make the right choice here.

Reply:
Ref: https://github.com/hashgraph/hedera-services/pull/14832#event-13948835760